### PR TITLE
fix: issue where if you set one geth network URI, all other default network URIs would vanish

### PR DESCRIPTION
**Pull request by unparalleled-js**
### What I did

Issue where if you set a network default URI for the geth plugin in your project, that the default network URIs for the other networks will now all missing from the config, so you get `KeyError` when trying to access them.

### How I did it

Use default URI when network config missing.

### How to verify it

Use `local` network in a project that specified the `mainnet` URI for the geth provide.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by fubuloubu** [comment]
related to #514 ?

**Review by fubuloubu — APPROVED** [review]
No test for this I suppose

**Comment by unparalleled-js** [comment]
> No test for this I suppose

I was able to add a test

File: tests/integration/cli/test_networks.py [hunk 1]
```diff
@@ -98,6 +101,11 @@ def test_list_yaml(ape_cli, runner):
...
 
 @skip_projects_except(["geth"])
-def test_change_default_from_config_file(ape_cli, runner):
+def test_geth(ape_cli, runner, networks):
     result = runner.invoke(ape_cli, ["networks", "list"])
     assert_rich_text(result.output, _GETH_NETWORKS_YAML)
+
+    # Assert that URI still exists for local network
+    # (was bug where one network's URI disappeared when setting different network's URI)
+    geth_provider = networks.get_provider_from_choice(f"ethereum:{LOCAL_NETWORK_NAME}:geth")
```
Comment (unparalleled-js) [inline tests/integration/cli/test_networks.py:110]: This part of the test requires the use of the `projects/` but is not CLI related..........

File: src/ape_geth/providers.py [hunk 2]
```diff
@@ -139,7 +139,9 @@ class GethProvider(Web3Provider, UpstreamProvider):
 
     @property
     def uri(self) -> str:
-        return getattr(self.config, self.network.ecosystem.name)[self.network.name]["uri"]
+        ecosystem_config = getattr(self.config, self.network.ecosystem.name)
+        network_config = ecosystem_config.get(self.network.name, DEFAULT_SETTINGS)
+        return network_config.get("uri", DEFAULT_SETTINGS["uri"])
```
Comment (fubuloubu) [inline src/ape_geth/providers.py:144]: Lol, using advice you gave in another PR against you:
```suggestion
        network_config = ecosystem_config.get(self.network.name) or DEFAULT_SETTINGS
        return network_config.get("uri") or DEFAULT_SETTINGS["uri"]
```

**Review by fubuloubu — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**